RecoPixelVertexing/PixelTrackFitting/src/PixelFitterByHelixProjections.cc

```diff
@@ -104,36 +104,20 @@ reco::Track* PixelFitterByHelixProjections::run(
   int nhits = hits.size();
   if (nhits <2) return 0;
 
-  vector<GlobalPoint> points(nhits);
-  vector<GlobalError> errors(nhits);
-  vector<bool> isBarrel(nhits);
-  
-  if (theTrackerWatcher.check(es)) {
-    edm::ESHandle<TrackerGeometry> trackerESH;
-    es.get<TrackerDigiGeometryRecord>().get(trackerESH);
-    theTracker = trackerESH.product();
-  }
-
-  if (theFieldWatcher.check(es)) {
+  if (!theField) { 
     edm::ESHandle<MagneticField> fieldESH;
     es.get<IdealMagneticFieldRecord>().get(fieldESH);
     theField = fieldESH.product();
```

Comment:
I'm mostly worried about the reallocation of the ES product.
Although changes from 3.8T to 0T in rereco/reprocessing applications are technically possible in the same job.

I can't recall the issue of "unexpected" relocation happening for the magnetic field, but it did for the geometry case I mentioned, see the fix in https://github.com/cms-sw/cmssw/pull/11417

@Dr15Jones @mmusich @diguida 
How "normal" is it for ES products to reallocate or change in memory otherwise when no actual change of values is expected?